main.go

```diff
@@ -63,6 +58,30 @@ func loadConfig(bidderInfos config.BidderInfos) (*config.Configuration, error) {
 	return config.New(v, bidderInfos, openrtb_ext.NormalizeBidderName)
 }
 
+func setGarbageCollectionConfig(cfg *config.Configuration) {
+	// Create a soft memory limit on the total amount of memory that PBS uses to tune the behavior
+	// of the Go garbage collector. In summary, `cfg.GarbageCollectorThreshold` serves as a fixed cost
+	// of memory that is going to be held garbage before a garbage collection cycle is triggered.
+	// This amount of virtual memory won’t translate into physical memory allocation unless we attempt
+	// to read or write to the slice below, which PBS will not do.
+	//
+	// Note that since Go 1.19, the GOMEMLIMIT environment variable can be used to achieve the same, when
+	// used in conjunction with GOGC.
+	// PBS supports that with the `go_runtime.soft_memory_limit` and `go_runtime.gc_percent` configuration
+	// options and this code will be removed once the deprecated `garbage_collector_threshold` field is removed.
+	if cfg.GarbageCollectorThreshold > 0 {
+		garbageCollectionThreshold := make([]byte, cfg.GarbageCollectorThreshold)
+		defer runtime.KeepAlive(garbageCollectionThreshold)
+	}
```

Comment:
Looking at the `KeepAlive` documentation:
```
460 // KeepAlive marks its argument as currently reachable.
461 // This ensures that the object is not freed, and its finalizer is not run,
462 // before the point in the program where KeepAlive is called.
/opt/homebrew/Cellar/go/1.19.4/libexec/src/runtime/mfinal.go
```
I don't think the `KeepAlive()` call, that with this `defer` statement will happen at the end of `setGarbageCollectionConfig()` and not at the end of `main()`, will have the intended effect of not freeing the `garbageCollectionThreshold` variable until PBS stops. This approach works in the current version of PBS because the `defer KeepAlive()` statement is found in `main()` instead. I believe we should put this call back in `main()` in order to have the intended effect or, if we want `setGarbageCollectionConfig()` to take care of the Garbage Collector config, we should make `garbageCollectionThreshold` a global variable again and initialize it here:
```
72     if cfg.GarbageCollectorThreshold > 0 {
73 -       garbageCollectionThreshold := make([]byte, cfg.GarbageCollectorThreshold)
74 -       defer runtime.KeepAlive(garbageCollectionThreshold)
   +       // garbageCollectionThreshold is declared at the top of main.go as a global variable
   +       garbageCollectionThreshold = make([]byte, cfg.GarbageCollectorThreshold)
75     }
main.go
```

Thoughts?